module/applications/actor/base-sheet.mjs

```diff
@@ -1115,6 +1135,39 @@ export default class ActorSheet5e extends ActorSheet {
 
   /* -------------------------------------------- */
 
+  /**
+   * Adjust the preparation mode of a dropped spell depending on the drop location on the sheet.
+   * @param {object} itemData    The item data requested for creation. **Will be mutated.**
+   */
+  _onDropSpell(itemData) {
+    if ( itemData.system.level == 0 || !["npc", "character"].includes(this.document.type) ) return;
+
+    // Determine the section it is dropped on, if any.
+    const mode = this._event.target.closest("div[data-type='spell']")?.dataset;
+
+    // If the spell is dropped on a section, set the mode to be that type.
+    if ( mode && mode.level > 0 ) {
+      itemData.system.preparation.mode = mode["preparation.mode"];
+    }
+
+    // If the spell is not dropped on a section, and on an NPC with no caster levels, set the spell to be innate.
+    else if ( this.document.type === "npc" && !this.document.system.details.spellLevel ) {
+      itemData.system.preparation.mode = "innate";
+    }
+
+    // If the spell is dropped on an actor with only pact magic (and other classes having 'none'), set the spell to be pact.
```

Comment:
Check the line length there, a bit longer than 120 characters. The parenthetical seems redundant here.

```suggestion
    // If the spell is dropped on an actor with only pact magic set the spell to be pact.
```